Model/FinalizeUnhandledOrders.php

```diff
@@ -0,0 +1,196 @@
+     * @param \Satispay\Satispay\Model\Method\Satispay $satispay
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     * @param \Satispay\Satispay\Model\Config $config
+     * @param FinalizePayment $finalizePaymentService
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
+        \Magento\Sales\Api\OrderRepositoryInterface $orderRepository,
+        \Magento\Sales\Api\OrderStatusHistoryRepositoryInterface $orderStatusRepository,
+        \Magento\Framework\Api\SearchCriteriaBuilder $searchCriteriaBuilder,
+        \Satispay\Satispay\Model\Method\Satispay $satispay,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Satispay\Satispay\Model\Config $config,
+        \Satispay\Satispay\Model\FinalizePayment $finalizePaymentService,
+        LoggerInterface $logger
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->orderStatusRepository = $orderStatusRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->storeManager = $storeManager;
+        $this->config = $config;
+        $this->finalizePaymentService = $finalizePaymentService;
+        $this->logger = $logger;
+    }
+
+    /**
+     *  Get list of orders from available stores and process them
+     */
+    public function finalizeUnhandledOrders()
+    {
+        $availableStores = $this->getAvailableStores();
+
+        foreach ($availableStores as $storeId) {
+            $rangeStart = $this->getStartDateScheduledTime($storeId);
+            $rangeEnd = $this->getEndDateScheduledTime();
+
+            $searchCriteria = $this->searchCriteriaBuilder
+                ->addFilter('state', \Magento\Sales\Model\Order::STATE_NEW)
+                ->addFilter('store_id', $storeId)
+                ->addFilter('updated_at', $rangeStart, 'gteq')
+                ->addFilter('updated_at', $rangeEnd, 'lteq')
+                ->create();
+
+            $orders = $this->orderRepository->getList($searchCriteria);
+
+            /** @var \Magento\Sales\Model\Order $order */
+            foreach ($orders->getItems() as $order) {
+                $orderPayment = $order->getPayment();
+                if (isset($orderPayment) && $orderPayment->getMethod() === 'satispay') {
+                    try {
+                        $this->processOrder($order);
+                    } catch (\Exception $e) {
+                        $orderId = $order->getEntityId();
+                        $this->logger->error("Could not finalize Order $orderId for Satispay payment: " . $e->getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param \Magento\Sales\Model\Order $order
+     */
+    private function processOrder(\Magento\Sales\Model\Order $order)
+    {
+        $orderId = $order->getEntityId();
+        $payment = $order->getPayment();
```

Comment:
check for payment not being null

Reply:
Payment cannot be null following this check `if (isset($orderPayment) && $orderPayment->getMethod() === 'satispay')` before function call